backend-lib/src/main/rust/lib.rs

```diff
@@ -726,27 +657,20 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_ge
         let db_data = wallet_db(&env, network, db_data)?;
         let account = AccountId::from(u32::try_from(accountj)?);
 
-        // We query the unverified balance including unmined transactions. Shielded notes
-        // in unmined transactions are never spendable, but this ensures that the balance
-        // reported to users does not drop temporarily in a way that they don't expect.
-        // `getVerifiedBalance` requires `ANCHOR_OFFSET` confirmations, which means it
-        // always shows a spendable balance.
-        let min_confirmations = NonZeroU32::new(1).unwrap();
-
-        (&db_data)
-            .get_target_and_anchor_heights(min_confirmations)
-            .map_err(|e| format_err!("Error while fetching anchor height: {}", e))
-            .and_then(|opt_anchor| {
-                opt_anchor
-                    .map(|(_, a)| a + 1)
-                    .ok_or(format_err!("Anchor height not available; scan required."))
-            })
-            .and_then(|anchor| {
-                (&db_data)
-                    .get_balance_at(account, anchor)
-                    .map_err(|e| format_err!("Error while fetching verified balance: {}", e))
-            })
-            .map(|amount| amount.into())
+        if let Some(wallet_summary) = db_data
+            .get_wallet_summary(0)
+            .map_err(|e| format_err!("Error while fetching balance: {}", e))?
+        {
+            wallet_summary
+                .account_balances()
+                .get(&account)
+                .ok_or_else(|| format_err!("Unknown account"))
+                .map(|balances| Amount::from(balances.sapling_balance.total()).into())
```

Comment:
Shouldn't this be `balance.total()` if it's intended for use as the top-line value? Or is the meaning of this API really `getShieldedBalance`?

Reply:
I thought this API was just the shielded balance (i.e. the balance that can be spent to recipients in our wallet), given that there are existing separate `getTotalTransparentBalance` and `getVerifiedTransparentBalance` APIs.

Reply:
Okay, yeah, as far as maintaining the current API goes I agree that this is the correct change. In the future, if we keep this API and don't replace it entirely with a wallet summary API, it should probably be renamed `getShieldedBalance` or something of the sort.